### PR TITLE
Adapt provisioner for running local

### PR DESCRIPTION
**Pull request by VOID404**
**Description**

This PR will make it possible, to run provisioner locally (for debugging, testing etc)

Changes proposed in this pull request:

- Change director config from reading a secret directly to mounting a file

**Related issue(s)**
https://github.com/kyma-project/test-infra/issues/7662

### CODE REVIEW

**Comment by VOID404** [comment]
This PR will require some changes in management plane config
/hold

**Review by Disper — CHANGES_REQUESTED** [review]
Please remember about updating https://github.com/kyma-project/control-plane/blob/main/components/provisioner/README.md

**Review by mvshao — CHANGES_REQUESTED** [review]
Please bump the Provisioner images

**Review by mvshao — COMMENTED** [review]
Thanks for bumping the image!

**Review by Disper — APPROVED** [review]
I see that this PR was attached to another one when I left some comments https://github.com/kyma-project/control-plane/pull/2886#pullrequestreview-1534314621. The comments there related to those parts are not a request for changes so I'm giving an LGTM here to make it easier to work on the other PR.

File: components/provisioner/README.md [hunk 1]
```diff
@@ -14,7 +14,9 @@ Before you can run Runtime Provisioner, you have to configure access to the Post
 $ docker run --rm -p 5432:5432 -e POSTGRES_PASSWORD=password postgres
 ```
 
-Runtime Provisioner also needs access to the cluster from which it fetches Secrets.  
+Afterwards the database has to be created and migrations have to be run.
```
Comment (grego952) [inline components/provisioner/README.md:17]:
```suggestion
Afterwards, the database must be created, and migrations must be run.
```

File: components/provisioner/README.md [hunk 2]
```diff
@@ -30,37 +32,47 @@ For tests to run properly, update the database schema in `./assets/database/prov
...
+| APP_DIRECTOR_URL                                              | Director URL                                                                                              | `http://compass-director.compass-system.svc.cluster.local:3000/graphql` |
+| APP_DOWNLOAD_PRE_RELEASES                                     |                                                                                                           | `true`                                                                  |
+| APP_ENQUEUE_IN_PROGRESS_OPERATIONS                            | Specifies whether operations in the `InProgress` state should be enqueued on the application startup      | `true`                                                                  |
+| APP_GARDENER_AUDIT_LOGS_POLICY_CONFIG_MAP                     | Name of the Config Map containing the audit logs policy                                                   | optional                                                                |
```
Comment (grego952) [inline components/provisioner/README.md:61]:
```suggestion
| APP_GARDENER_AUDIT_LOGS_POLICY_CONFIG_MAP                     | Name of the ConfigMap containing the audit logs policy                                                   | optional                                                                |
```

File: components/provisioner/README.md [hunk 3]
```diff
@@ -14,7 +14,9 @@ Before you can run Runtime Provisioner, you have to configure access to the Post
 $ docker run --rm -p 5432:5432 -e POSTGRES_PASSWORD=password postgres
 ```
 
-Runtime Provisioner also needs access to the cluster from which it fetches Secrets.  
+Afterwards the database has to be created and migrations have to be run.
+
+Runtime Provisioner also needs a kubeconfig for a garden and auth data for Director.
```
[the remark this answers is not in the document]
Comment (VOID404) [reply components/provisioner/README.md:19]: Both should be understandable. 
Gardener garden, where seeds are planted and shoots grow ;)

**Review by mvshao — APPROVED** [review]

**Review by mvshao — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**